cpp/benchmarks/join/join_benchmark.cu

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <vector>
+#include <cudf/cudf.h>
+#include <utilities/error_utils.hpp>
+#include <tests/utilities/column_wrapper.cuh>
+
+#include "generate_input_tables.cuh"
+
+
+template<typename key_type, typename payload_type>
+static void join_benchmark(benchmark::State& state)
+{
+    const gdf_size_type build_table_size {(gdf_size_type) state.range(0)};
+    const gdf_size_type probe_table_size {(gdf_size_type) state.range(1)};
+    const gdf_size_type rand_max_val {build_table_size * 2};
+    const double selectivity = 0.3;
+    const bool is_build_table_key_unique = true;
+
+    // Generate build and probe table data
+
+    auto build_key_column     = new cudf::test::column_wrapper<key_type>(build_table_size);
+    auto build_payload_column = new cudf::test::column_wrapper<payload_type>(build_table_size);
+    auto probe_key_column     = new cudf::test::column_wrapper<key_type>(probe_table_size);
+    auto probe_payload_column = new cudf::test::column_wrapper<payload_type>(probe_table_size);
+
+    generate_input_tables<key_type, gdf_size_type>(
+        (key_type *)build_key_column->get()->data, build_table_size,
+        (key_type *)probe_key_column->get()->data, probe_table_size,
+        selectivity, rand_max_val, is_build_table_key_unique
+    );
+
+    linear_sequence<payload_type, gdf_size_type><<<(build_table_size+127)/128,128>>>(
```

Comment:
Just pass a value initializer lambda into the `column_wrapper` constructor instead of doing this. 

`cudf::test::column_wrapper<payload_type>(probe_table_size, [](gdf_index_type row_index){return row_index;});`

This will accomplish the same result.

Reply:
Switched to column_wrapper in https://github.com/rapidsai/cudf/pull/2105/commits/c3a81e3fd31cfe9b01d6ab580dd9ee304c1b86de